core-api/src/main/java/com/optimizely/ab/event/internal/EventBuilderV1.java

```diff
@@ -0,0 +1,274 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.event.internal;
+
+import com.optimizely.ab.event.LogEvent;
+import com.optimizely.ab.internal.ProjectValidationUtils;
+import com.optimizely.ab.bucketing.Bucketer;
+import com.optimizely.ab.config.Attribute;
+import com.optimizely.ab.config.EventType;
+import com.optimizely.ab.config.Experiment;
+import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.config.Variation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import static com.optimizely.ab.event.LogEvent.RequestMethod;
+
+/**
+ * Event builder that produces
+ * <a href="https://help.optimizely.com/hc/en-us/articles/200040195-Tracking-offline-conversion-events-with-Optimizely">
+ *     Optimizely offline conversion</a> events.
+ */
+public class EventBuilderV1 extends EventBuilder {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventBuilderV1.class);
+
+    private static final String ENDPOINT_FORMAT = "https://%s.log.optimizely.com/event";
+
+    // offline conversion parameter names and prefixes
+    private static final String ACCOUNT_ID_PARAM = "d";
+    private static final String BUILD_VERSION = "java-sdk-" + BuildVersionInfo.VERSION;
+    private static final String EMPTY_BODY = "";
+    private static final String EXPERIMENT_PARAM_PREFIX = "x";
+    private static final String GOAL_ID_PARAM = "g";
+    private static final String GOAL_NAME_PARAM = "n";
+    private static final String PPID_PARAM = "p";
+    private static final String PROJECT_ID_PARAM = "a";
+    private static final String EVENT_VALUE_PARAM = "v";
+    private static final String SEGMENT_PARAM_PREFIX = "s";
+    private static final String SOURCE_PARAM = "src";
+    private static final String TIME_PARAM = "time";
+    private static final String USER_ID_PARAM = "u";
+
+    public LogEvent createImpressionEvent(@Nonnull ProjectConfig projectConfig,
+                                          @Nonnull Experiment activatedExperiment,
+                                          @Nonnull Variation variation,
+                                          @Nonnull String userId,
+                                          @Nonnull Map<String, String> attributes) {
+
+        Map<String, String> requestParams = new HashMap<String, String>();
+        addCommonRequestParams(requestParams, projectConfig, userId, attributes);
+        addImpressionGoal(requestParams, activatedExperiment);
+        addExperiment(requestParams, activatedExperiment, variation);
+
+        return new LogEvent(RequestMethod.GET,
```

Comment:
We should make LogEvent as lightweight as possible since developers will be using it and we want to abstract event creation away from them. In a future change, instead of passing around a `Map` we can use `putAll` to combine maps returned by helper functions in the root creation method.